### ISSUE TITLE
Count-lock does not lock rocket count while flying with elytra

### ISSUE THREAD

**Issue by Rick-Thompson** [RockinChaos/ItemJoin#308]
I have rockets count-locked at 64. They refill when launched in the air while standing on the ground but when flying they do not refill.

I am running build 471 but it still has the same issue on the older builds. Running on server spigot 1.16.1

My items.yml

```
#// <>==<>==<> ItemJoin's Custom Items On Join, by RockinChaos <>==<>==<> //#
#// See the Documentations page for a more in-depth tutorial; https://www.spigotmc.org/wiki/itemjoin/ //#
items-Version: 6
items-Delay: 2
items-Overwrite: false
items-Spamming: false
items-RestrictCount : true

items:
    elytra-item:
      id: ELYTRA
      slot: Arbitrary
      triggers: join, respawn, world-change
      itemflags: unbreakable
    food-item:
      id: bread
      slot: Arbitrary
      count: 16
      itemflags: first-join
    rocket-item:
      id: FIREWORK_ROCKET
      slot: Arbitrary
      itemflags: count-lock
      count: 64
      triggers: join, respawn, world-change
```

**Comment by RockinChaos** [comment]
Replicated, fixed and tested as working in the latest snapshot; http://ci.craftationgaming.com/job/ItemJoin/472/

I was originally like, wait a minute... how can you launch a firework without clicking a block... *clicks air* hmmm nothing is happening. *Tries to launch firework with elytra while flying* *WOOOOOOOOOOOOOWWWW THIS IS AMAZING!* Sooo yeah you taught me a feature I had no idea existed. That's a first for me xD. Totally curious when this was introduced because if it existed when elytra's were added I am going to bang my head into a desk.

**Comment by Rick-Thompson** [comment]
It was added in the minor update right after they added the elytra. I think it was 1.11.1 that it was added but I could be wrong. I just loaded #472 and it is working good now, Thanks!!

**Comment by RockinChaos** [comment]
This bug has been officially fixed in the release of ItemJoin v5.0.7.
You can grab it here; Download: https://www.spigotmc.org/resources/itemjoin.12661/download?version=354306

Please read the changelog as the documentation has not been updated yet;
https://github.com/RockinChaos/ItemJoin/wiki/Recent-Changes

**Comment by PaBox** [comment]
Coming relatively late to this conversation, but still wanted to add it - there is no current example of it in the documentation and no issue regarding this problem: When limiting the use of firework rockets the warmup/cooldown does not apply, if used with an equipped elytra. I guess because a different function is called when clicking air and flying - I am therefore not able to limit the use of fireworks (while flying). This is in v6.0.2 on Paper 1.18.2 and has been a problem since at least 2 years. This should be a thing, as the messages/commands are also not told/executed which are associated with the firework, it feels overlooked.
